cohort-engine-api-web/src/test/java/com/ibm/cohort/engine/api/service/vt/DefaultVT.java

```diff
@@ -376,9 +383,71 @@ public void testXMLUnitAssertXPathValues() throws Exception {
         assertXpathValuesNotEqual("//jvm[2]/@current", "//jvm[3]/@current", myJavaFlavours);
     }
 
+    @Test
+	public void testValueSetUpload(){
+		final String RESOURCE = getUrlBase() + "/{version}/valueset";
+		Assume.assumeTrue(isServiceDarkFeatureEnabled(CohortEngineRestConstants.DARK_LAUNCHED_VALUE_SET_UPLOAD));
+
+		String auth = dataServerConfig.getUser() + ":" + dataServerConfig.getPassword();
+		byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+
+		RequestSpecification request = buildBaseRequest(new Headers(new Header("Authorization", "Basic " + new String(encodedAuth))))
+				.param("version", ServiceBuildConstants.DATE)
+				.queryParam("fhir_server_rest_endpoint", dataServerConfig.getEndpoint())
+				.queryParam("updateIfExists", false)
+				.multiPart(CohortEngineRestHandler.VALUE_SET_PART, new File("src/test/resources/2.16.840.1.113762.1.4.1114.7.xlsx"));
+
+		ValidatableResponse response = request.post(RESOURCE, getServiceVersion()).then();
+		runSuccessValidation(response, ContentType.JSON, HttpStatus.SC_OK);
+	}
+
+	@Test
+	public void testValueSetAlreadyExists(){
+		testValueSetUpload();
+		final String RESOURCE = getUrlBase() + "/{version}/valueset";
+
+		String auth = dataServerConfig.getUser() + ":" + dataServerConfig.getPassword();
+		byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+
+		RequestSpecification request = buildBaseRequest(new Headers(new Header("Authorization", "Basic " + new String(encodedAuth))))
+				.param("version", ServiceBuildConstants.DATE)
+				.queryParam("fhir_server_rest_endpoint", dataServerConfig.getEndpoint())
+				.queryParam("updateIfExists", false)
+				.multiPart(CohortEngineRestHandler.VALUE_SET_PART, new File("src/test/resources/2.16.840.1.113762.1.4.1114.7.xlsx"));
+
+		ValidatableResponse response = request.post(RESOURCE, getServiceVersion()).then();
+		runSuccessValidation(response, ContentType.JSON, 409);
```

Comment:
Can `409` be `HttpStatus.SC_CONFLICT` here?

Reply:
Definitely, will change

Reply:
Oh, I thought that constant would work since it's from the same place as `HttpStatus.SC_OK` used in some of the other tests.

Reply:
I misread your import, that does work, I'll make that change.